src/main/kotlin/io/github/inductiveautomation/kindling/core/Kindling.kt

```diff
@@ -3,116 +3,250 @@ package io.github.inductiveautomation.kindling.core
+            override val displayName: String = "UI"
+            override val preferences: List<Preference<*>> = listOf(Theme, ScaleFactor)
+        }
+
+        object Advanced : PreferenceCategory {
+            val Debug: Preference<Boolean> = preference(
+                name = "Debug Mode",
+                description = null,
+                default = false,
+                editor = {
+                    PreferenceCheckbox("Enable debug features")
+                },
+            )
+
+            val HyperlinkStrategy: Preference<LinkHandlingStrategy> = preference(
+                name = "Hyperlink Strategy",
+                default = LinkHandlingStrategy.OpenInBrowser,
+                serializer = LinkHandlingStrategy.serializer(),
+                editor = {
+                    JComboBox(LinkHandlingStrategy.values()).apply {
+                        selectedItem = currentValue
 
-    object Advanced : PreferenceCategory() {
-        val Debug: Preference<Boolean> by preference(
-            name = "Debug Mode",
-            default = false,
-            editor = {
-                JCheckBox("Enable debug features").apply {
-                    isSelected = currentValue
-                    addItemListener { e ->
-                        currentValue = e.stateChange == ItemEvent.SELECTED
+                        configureCellRenderer { _, value, _, _, _ ->
+                            text = value?.description
+                        }
+
+                        addActionListener {
+                            currentValue = selectedItem as LinkHandlingStrategy
+                        }
                     }
-                }
-            },
-        )
+                },
+            )
 
-        override fun toString(): String = "Advanced"
-    }
+            override val displayName: String = "Advanced"
+            override val preferences: List<Preference<*>> = listOf(Debug, HyperlinkStrategy)
+        }
 
-    val preferenceCategories: List<PreferenceCategory> = listOf(General, UI, Advanced)
+        private val preferencesPath: Path = Path(System.getProperty("user.home"), ".kindling").also {
+            it.createDirectories()
+        }.resolve("preferences.json")
+
+        private val preferencesJson = Json {
+            ignoreUnknownKeys = true
+            prettyPrint = true
+        }
+
+        val categories: List<PreferenceCategory> = buildList {
+            add(General)
+            add(UI)
+            addAll(Tool.tools.filterIsInstance<PreferenceCategory>())
+            // put advanced last
+            add(Advanced)
+        }
+
+        private val internalState: MutableMap<String, MutableMap<String, JsonElement>> = try {
+            // try to deserialize from file
+            preferencesPath.inputStream().use { inputStream ->
+                @OptIn(ExperimentalSerializationApi::class)
+                preferencesJson.decodeFromStream(inputStream)
+            }
+        } catch (e: Exception) {
```

Comment:
It would be nice if this failure could be made more granular - if it's valid JSON, but one of the keys is wrong or bad in some way, it would be nice if we could still accept any other, valid stored preferences